manual/src/cmds/profile-perf.etex

```diff
@@ -0,0 +1,90 @@
+\chapter{Profiling (perf)} \label{c:profiler-perf}
+%HEVEA\cutname{profiler-perf.html}
+
+This chapter describes how to use \texttt{perf}, the Linux Performance Events tool, to profile OCaml programs.
+
+Linux Performance Events (\texttt{perf (1)}) is a set of tools for performance observability, including CPU performance counter profiling and static and dynamic tracing. This chapter covers key features such as \texttt{perf-record(1)} for recording events and \texttt{perf-report(1)} for printing and visualising recorded events. \textt{perf} has many other features for profiling and visualising performance, see (man perf(1), \href{https://perfwiki.github.io/main/}{Perf wiki}, or \href{https://www.brendangregg.com/perf.html}{Brendan Gregg's Blog}) for more general documentation.
+
+\section{s:ocamlperf-compiling}{Compiling for Profiling}
+
+The process for compiling an executable for time profiling depends on the OCaml version. For OCaml versions 4.14 and earlier, either frame pointers or DWARF can be used, while for OCaml 5.0 and later, enabling frame pointers is recommended.
+
+To enable frame pointers, configure the compiler with \texttt{--enable-frame-pointers}. Alternatively, you can install an opam switch with frame pointers enabled, as follows:
+
+\begin{verbatim}
+        opam switch create <YOUR-SWITCH-NAME-HERE> ocaml-option-fp
+\end{verbatim}
+
+Frame pointer support for OCaml is available on x86_64 architecture on Linux starting with OCaml 4.12 and on macOS from OCaml 5.3. ARM64 architecture is supported on Linux and macOS from OCaml 5.4, while other Tier-1 architectures (POWER, RISC-V, and s390x) are currently unsupported.
+
+OCaml 5 requires frame pointers due to its use of non-contiguous stacks for effects, as mentioned in \href{https://dl.acm.org/doi/10.1145/3453483.3454039}{Retrofitting Effect Handlers Onto OCaml} (Section 5.5). Non-contiguous stacks are incompatible with \texttt{perf}’s stack-copying approach, so traces produced for OCaml 5 without frame pointers will be truncated and inaccurate. Profiling will also be truncated if the Linux distribution doesn't enable frame pointers for libraries that OCaml links against.
+
+\section{s:ocamlperf-profiling}{Profiling an Execution}
+
+The basic \texttt{perf} command is:
+\begin{verbatim}
+        perf record -F 99 --call-graph fp <YOUR_EXECUTABLE>
+\end{verbatim}
+
+The \texttt{-F 99} option tells \texttt{perf} to sample at 99Hz, which avoids generating excessive data for longer runs and is unlikely to be in lockstep with other periodic activities. The \texttt{--call-graph fp} instructs \texttt{perf} to use frame pointers to get the call-graph, followed by the OCaml executable you want to profile. This command creates a \texttt{perf.data} file in the current directory. Alternatively, use \texttt{--output} to choose a more descriptive filename.
```

Comment:
```suggestion
The \texttt{-F 99} option tells \texttt{perf} to sample at 99Hz, which avoids generating excessive data for longer runs and minimising overlap with other periodic activities. The \texttt{--call-graph fp} instructs \texttt{perf} to use frame pointers to get the \texttt{call-graph}, followed by the OCaml executable you want to profile. This command creates a \texttt{perf.data} file in the current directory. Alternatively, use \texttt{--output} to choose a more descriptive filename.
```

Does "minimising overlap" still say what you wanted to say? If so, this syntax is a little tighter.

If `call-graph` isn't in monospace as suggested, do you mean "call graph" in a more general sense rather than the command? If so, then remove the hyphen.

Reply:
Yes it means the same thing. 

The aim is to avoid running the sampling in such a way that it always runs at the same time as another periodic activity or it always misses an interesting task by running when that task is definitely not running.